### ISSUE TITLE
check image built locally without pulling it from remote registry

### ISSUE THREAD

**Issue by maratsal** [redhat-openshift-ecosystem/openshift-preflight#593]
## Bug Description
Not sure if this is an issue, maybe feature request. When I build container locally and try to use preflight to check compliance it is trying to pull container image from the remote registry with prepending docker.io. Is there a way to do check up of locally available containers? Here is an example:
```
$ preflight check container mycontainer:mytag
time="2022-05-02T14:51:59-04:00" level=info msg="certification library version 0.0.0 <commit: 100e67e10e75507e91dbb7b2cc86db5651ea5a73>"
Error: failed to pull remote container: GET https://index.docker.io/v2/library/mycontainer/manifests/mytag: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/mycontainer Type:repository]]
time="2022-05-02T14:52:00-04:00" level=fatal msg="failed to pull remote container: GET https://index.docker.io/v2/library/mycontainer/manifests/mytag: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/mycontainer Type:repository]]"
```

## Version and Command Invocation
(The output of `preflight --version`)
```
preflight --version
preflight version 0.0.0 <commit: 100e67e10e75507e91dbb7b2cc86db5651ea5a73>
```
## Steps to Reproduce:
(How can we reproduce this?)
1) docker build -t  mycontainer:mytag .
2) preflight check container mycontainer:mytag

## Expected Result
(What did you expect to happen and why?)
I would expect it to try local images and then if it will not find any go through all available repositories.

## Actual Result
(What actually happened)
We get error message.

## Additional Context
(Anything else you think might help us troubleshoot, like your platform, dependency versions, etc).
This will help to avoid pushing non-compliant images to the registry and also do local tests as a developer.

**Comment by bcrochet** [comment]
At this point, this would be an RFE. Preflight makes an assumption that the image under test has been pushed to a remote registry somewhere.

**Comment by rocrisp** [comment]
What do you think about adding a boolean flag --uselocalimage.
The side effect is --submit will not be available.
For example, the command is something like this.
./preflight check container --uselocalimage <localImage>

**Comment by bcrochet** [comment]
It's not quite as simple as that. One workaround would be to fire up a local-to-your-machine registry, and store it there.

**Comment by maratsal** [comment]
> It's not quite as simple as that. One workaround would be to fire up a local-to-your-machine registry, and store it there.

I used this one - the problem here with unsecure registry. I had to tweak preflight tool code a bit to be able to work with such registry. Added following line `options = append(options, crane.Insecure)` here https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/internal/engine/engine.go#L63 to do that.

And I could not make it work with registry with self signed certificate. Some option like --tls-verify=false would be good.

But I would still like to see some option like suggested by @rocrisp
